lxd/certificates.go

```diff
@@ -298,10 +297,24 @@ func updateCertificateCacheFromLocal(d *Daemon) error {
 			continue
 		}
 
-		newCerts[dbCert.Type][shared.CertFingerprint(cert)] = *cert
+		id, err := dbCert.ToIdentity()
+		if err != nil {
+			logger.Warn("Failed to convert node certificate into identity entry", logger.Ctx{"error": err})
+			continue
+		}
+
+		identityCacheEntries = append(identityCacheEntries, identity.CacheEntry{
+			Identifier:           id.Identifier,
+			AuthenticationMethod: string(id.AuthMethod),
+			IdentityType:         string(id.Type),
+			Certificate:          cert,
+		})
 	}
 
-	d.clientCerts.SetCertificates(newCerts)
+	err = d.identityCache.ReplaceAll(identityCacheEntries)
+	if err != nil {
+		return fmt.Errorf("Failed to update identity cache: %w", err)
```

Comment:
Can we differentiate this error from the same one in updateCertificateCache?

Reply:
Missed this one. Fixed now.